<?xml version="1.0"?>
<launch>

	<!-- arguments -->
	<arg name="domain_path"	default="/mnt/c/Users/Public/ROSPlan/src/rosplan/rosplan_project/simple_harpia_dataset/p0/5_goals/domain.pddl" />
	<arg name="problem_path"	default="$(find rosplan_planning_system)/test/problem.pddl" />
	<arg name="data_path"            value="$(find rosplan_planning_system)/test/" />
    <arg name="action_dispatch_topic"	default="/rosplan_plan_dispatcher/action_dispatch" />
	<arg name="action_feedback_topic"	default="/rosplan_plan_dispatcher/action_feedback" />
	<arg name="hypothesis_goals"	default="/mnt/c/Users/Public/ROSPlan/src/rosplan/rosplan_project/simple_harpia_dataset/p2/5_goals/hyps.dat" />
	<arg name="temp_observations"	default="/mnt/c/Users/Public/ROSPlan/src/rosplan/rosplan_project/simple_harpia_dataset/p2/5_goals/temp_obs.dat" />
	<arg name="observations"	default="/mnt/c/Users/Public/ROSPlan/src/rosplan/rosplan_project/simple_harpia_dataset/p2/5_goals/obs.dat" />
	<arg name="initial_state"	default="/mnt/c/Users/Public/ROSPlan/src/rosplan/rosplan_project/simple_harpia_dataset/p2/5_goals/initial_state.pddl" />
	<arg name="log_file"	default="/mnt/c/Users/Public/ROSPlan/src/rosplan/rosplan_project/simple_harpia_dataset/p2/5_goals/goal_mirroring_log_file_lpg_all.txt" />
	<!-- <arg name="log_file"	default="/mnt/c/Users/Public/ROSPlan/src/rosplan/rosplan_project/simple_harpia_dataset/p2/5_goals/goal_mirroring_log_file_mean.txt" /> -->
	<!-- <arg name="log_file"	default="/mnt/c/Users/Public/ROSPlan/src/rosplan/rosplan_project/simple_harpia_dataset/p2/5_goals/goal_mirroring_log_file_min.txt" /> -->
	<arg name="real_goal"	default="/mnt/c/Users/Public/ROSPlan/src/rosplan/rosplan_project/simple_harpia_dataset/p2/5_goals/real_hyp.dat" />



    <!-- main coordinator -->
    <node pkg="rosplan_project" type="Diverse_Mirroring.py" name="coordinator" respawn="false" required="false" output="screen">
        <!-- <param name="max_prm_size"    value="$(arg max_prm_size)" /> -->
		<param name="planner_command"      value="timeout 60 $(find rosplan_planning_system)/common/bin/lpg-td -o DOMAIN -f PROBLEM -n 5 -seed 2023 -out $(find rosplan_planning_system)/test/lpgplan" />
        <!-- <param name="planner_command" value="timeout 10 $(find rosplan_planning_system)/common/bin/popf DOMAIN PROBLEM" /> -->
        <param name="domain_path"     value="$(arg domain_path)" />
        <param name="problem_path"    value="$(find rosplan_planning_system)/test/problem.pddl" />
        <param name="data_path"       value="$(arg data_path)" />
		<param name="hypothesis_goals"       value="$(arg hypothesis_goals)" />
		<param name="temp_observations"       value="$(arg temp_observations)" />
		<param name="observations"       value="$(arg observations)" />
		<param name="initial_state"       value="$(arg initial_state)" />
		<param name="log_file"       value="$(arg log_file)" />
		<param name="real_goal"       value="$(arg real_goal)" />
    </node>

	<!-- knowledge base -->
	<node name="rosplan_knowledge_base" pkg="rosplan_knowledge_base" type="knowledgeBase" respawn="false" output="screen">
		<param name="domain_path" value="$(arg domain_path)" />
		<param name="problem_path" value="/mnt/c/Users/Public/ROSPlan/src/rosplan/rosplan_project/simple_harpia_dataset/p2/5_goals/initial_state.pddl" />
		<!-- conditional planning flags -->
		<param name="use_unknowns" value="false" />
	</node>

	<!-- problem generation -->
	<include file="$(find rosplan_planning_system)/launch/includes/problem_interface.launch">
		<arg name="knowledge_base"   value="rosplan_knowledge_base" />
		<arg name="domain_path"      value="$(arg domain_path)" />
		<arg name="problem_path"     value="$(find rosplan_planning_system)/test/problem.pddl" />
		<arg name="problem_topic"    value="problem_instance" />
	</include>

	<!-- planner interface -->
	<include file="$(find rosplan_planning_system)/launch/includes/planner_interface_lpg.launch">
		<arg name="use_problem_topic"    value="true" />
		<arg name="problem_topic"        value="/rosplan_problem_interface/problem_instance" />
		<arg name="planner_topic"        value="planner_output" />
		<arg name="domain_path"          value="$(arg domain_path)" />
		<arg name="problem_path"         value="$(find rosplan_planning_system)/test/problem.pddl" />
		<arg name="data_path"            value="$(arg data_path)" />
		<arg name="planner_command"      value="timeout 60 $(find rosplan_planning_system)/common/bin/lpg-td -o DOMAIN -f PROBLEM -n 5 -seed 2023 -out $(find rosplan_planning_system)/test/lpgplan" />

	</include>

    <!-- plan parsing -->
	<node name="rosplan_parsing_interface" pkg="rosplan_planning_system" type="pddl_simple_plan_parser" respawn="false" output="screen">
		<param name="knowledge_base" value="rosplan_knowledge_base" />
		<param name="planner_topic"  value="/rosplan_planner_interface/planner_output" />
		<param name="plan_topic"     value="complete_plan" />
	</node>

	<!-- plan dispatching -->
	<node name="rosplan_plan_dispatcher" pkg="rosplan_planning_system" type="pddl_simple_plan_dispatcher" respawn="false" output="screen">
		<param name="knowledge_base"        value="rosplan_knowledge_base" />
		<param name="plan_topic"            value="/rosplan_parsing_interface/complete_plan" />
		<param name="action_dispatch_topic" value="action_dispatch" />
		<param name="action_feedback_topic" value="action_feedback" />
	</node>


    <!-- sim actions -->
	<include file="$(find rosplan_planning_system)/launch/includes/simulated_action.launch" >
		<arg name="pddl_action_name" value="go_to" />
	</include>
	<include file="$(find rosplan_planning_system)/launch/includes/simulated_action.launch" >
		<arg name="pddl_action_name" value="take_image" />
	</include>
	<include file="$(find rosplan_planning_system)/launch/includes/simulated_action.launch" >
		<arg name="pddl_action_name" value="pulverize_region" />
	</include>
	<include file="$(find rosplan_planning_system)/launch/includes/simulated_action.launch" >
		<arg name="pddl_action_name" value="recharge_battery" />
	</include>
    <include file="$(find rosplan_planning_system)/launch/includes/simulated_action.launch" >
		<arg name="pddl_action_name" value="recharge_input" />
	</include>
    
</launch>